services/subscriber/service.go

```diff
@@ -494,3 +562,69 @@ func (b *balancewriter) Statistics(tags map[string]string) []models.Statistic {
 	}
 	return statistics
 }
+
+type dbrp struct {
+	db string
+	rp string
+}
+
+// subscriptionRouter has a mutex lock on the hot path for database writes - make sure that the lock is very tight.
+type subscriptionRouter struct {
+	mu            sync.RWMutex
+	ready         bool
+	m             map[dbrp][]*chanWriter
+	writeFailures *int64
+	Logger        *zap.Logger
+}
+
+func newSubscriptionRouter(statistics *Statistics) *subscriptionRouter {
+	return &subscriptionRouter{
+		ready:         true,
+		writeFailures: &statistics.WriteFailures,
+		Logger:        zap.NewNop(),
```

Comment:
Looks like this is only ever a Nop? Can it be removed?

Reply:
We call `WithLogger` from `Server.Open`. We should probably call it in plutonium as well.